### ISSUE TITLE
Tree closed on web skia gold failure

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#93719]
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_tests_6/2815/overview

> Skia Gold received an unapproved image in post-submit 

/cc @yjbanov @Piinks @mdebbar 

```
09:24 +913 ~16: test/material/text_field_test.dart: text selection style 2                                                                                                                             [ +968 ms] Preparing to send command: {"imageFile":"/b/s/w/ir/x/t/flutter_tools.QSWAYH/flutter_web_platform.JYFBQM/imageYNXFIN/image","key":"text_field_golden.TextSelectionStyle.2.png","update":false}
[ +293 ms] <<< {"success":false,"message":"Exception: Skia Gold received an unapproved image in post-submit \ntesting. Golden file images in flutter/flutter are triaged \nin pre-submit during code review for the given PR.\n\nVisit https://flutter-gold.skia.org/ to view and approve \nthe image(s), or revert the associated change. For more \ninformation, visit the wiki: \nhttps://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter\n\nDebug information for Gold:\nstdout: Given image with hash 7fbe2f57b4d0038bba905ff114d3e61c for test material.text_field_golden.TextSelectionStyle.2\nExpectation for test: 3f0e2676b3bb8b8377134cec64e2650c (positive)\nExpectation for test: 7fbe2f57b4d0038bba905ff114d3e61c (negative)\nExpectation for test: ace796494b6ae4880df0e072a797efe2 (positive)\nExpectation for test: e89799314b9a9d6c32ac49970ee38cd0 (positive)\nExpectation for test: 160b763fcd8c8b04b873ca8e7077f3f5 (positive)\nExpectation for test: 2625392d4b3e1d5ffbf0bdf8e468ae10 (positive)\nExpectation for test: 391b2fe02fc4f172359fae4a6d1f4c8c (positive)\nExpectation for test: 96311ed4cfa913e2d65530081ef55e24 (positive)\nExpectation for test: 9ec587731d1f0b3c18aef2fff21dada1 (positive)\nExpectation for test: abaf6498750d31cecc246eedb37e7f65 (positive)\nExpectation for test: d0177da7b27bee5e29b85cc654926d7a (positive)\nExpectation for test: a13e2e3f609facb396859e9047c65355 (positive)\nExpectation for test: 19d52ac03f6f6264dc4c6a44f6e0b3a4 (positive)\nExpectation for test: 43d231f5a523688975dbaa2aba840852 (positive)\nExpectation for test: 448b783fda381ee64ec770ef127c493d (positive)\nExpectation for test: 6e76e4ca6b118afeed72fd8bbd44cc14 (positive)\nExpectation for test: 956892f2cd6572a97eafbf23d26e3ef9 (positive)\nExpectation for test: 99a6be150ce463a870b0c2f6cf38825e (positive)\nExpectation for test: a07a478bbc7b356282c3cfb8cf4d3cf7 (positive)\nExpectation for test: c42e8892f567727bc09df7871e1a9bc3 (positive)\nExpectation for test: dde577bff6ec924242f56fb78d3acff0 (positive)\nExpectation for test: fe5eff74cb98ff133029e6c89aa3601a (positive)\nExpectation for test: 966f047eb83a1405a512b2762923d708 (positive)\nExpectation for test: 9c200113aba5471b3214a26d3f991524 (positive)\nUntriaged or negative image: https://flutter-gold.skia.org/detail?test=material.text_field_golden.TextSelectionStyle.2&digest=7fbe2f57b4d0038bba905ff114d3e61c\n\nstderr: Test: material.text_field_golden.TextSelectionStyle.2 FAIL\n\n"}

09:25 +913 ~16: test/material/text_field_test.dart: text selection style 2                                                                                                                             
09:25 +913 ~16: test/material/text_field_test.dart: text selection style 2                                                                                                                             
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: one widget whose rasterized image matches golden image
"text_field_golden.TextSelectionStyle.2.png"
  Actual: _WidgetTypeFinder:<exactly one widget with type "MaterialApp" (ignoring offstage widgets):
MaterialApp(state: _MaterialAppState#1b275)>
   Which: Exception: Skia Gold received an unapproved image in post-submit
          testing. Golden file images in flutter/flutter are triaged
          in pre-submit during code review for the given PR.

          Visit https://flutter-gold.skia.org/ to view and approve
          the image(s), or revert the associated change. For more
          information, visit the wiki:
          https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter

          Debug information for Gold:
          stdout: Given image with hash 7fbe2f57b4d0038bba905ff114d3e61c for test
material.text_field_golden.TextSelectionStyle.2
          Expectation for test: 3f0e2676b3bb8b8377134cec64e2650c (positive)
          Expectation for test: 7fbe2f57b4d0038bba905ff114d3e61c (negative)
          Expectation for test: ace796494b6ae4880df0e072a797efe2 (positive)
          Expectation for test: e89799314b9a9d6c32ac49970ee38cd0 (positive)
          Expectation for test: 160b763fcd8c8b04b873ca8e7077f3f5 (positive)
          Expectation for test: 2625392d4b3e1d5ffbf0bdf8e468ae10 (positive)
          Expectation for test: 391b2fe02fc4f172359fae4a6d1f4c8c (positive)
          Expectation for test: 96311ed4cfa913e2d65530081ef55e24 (positive)
          Expectation for test: 9ec587731d1f0b3c18aef2fff21dada1 (positive)
          Expectation for test: abaf6498750d31cecc246eedb37e7f65 (positive)
          Expectation for test: d0177da7b27bee5e29b85cc654926d7a (positive)
          Expectation for test: a13e2e3f609facb396859e9047c65355 (positive)
          Expectation for test: 19d52ac03f6f6264dc4c6a44f6e0b3a4 (positive)
          Expectation for test: 43d231f5a523688975dbaa2aba840852 (positive)
          Expectation for test: 448b783fda381ee64ec770ef127c493d (positive)
          Expectation for test: 6e76e4ca6b118afeed72fd8bbd44cc14 (positive)
          Expectation for test: 956892f2cd6572a97eafbf23d26e3ef9 (positive)
          Expectation for test: 99a6be150ce463a870b0c2f6cf38825e (positive)
          Expectation for test: a07a478bbc7b356282c3cfb8cf4d3cf7 (positive)
          Expectation for test: c42e8892f567727bc09df7871e1a9bc3 (positive)
          Expectation for test: dde577bff6ec924242f56fb78d3acff0 (positive)
          Expectation for test: fe5eff74cb98ff133029e6c89aa3601a (positive)
          Expectation for test: 966f047eb83a1405a512b2762923d708 (positive)
          Expectation for test: 9c200113aba5471b3214a26d3f991524 (positive)
          Untriaged or negative image:
https://flutter-gold.skia.org/detail?test=material.text_field_golden.TextSelectionStyle.2&digest=7fbe2f57b4d0038bba905ff114d3e61c

          stderr: Test: material.text_field_golden.TextSelectionStyle.2 FAIL

When the exception was thrown, this was the stack:
../dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 251:49  throw_
../packages/test_api/src/expect/async_matcher.dart.js 145:22                     fail
../packages/test_api/src/expect/async_matcher.dart.js 120:18                     <fn>
../packages/stack_trace/src/stack_zone_specification.dart.js 160:98              <fn>
../packages/stack_trace/src/stack_zone_specification.dart.js 212:16              [_run]
../packages/stack_trace/src/stack_zone_specification.dart.js 160:80              <fn>
../dart-sdk/lib/async/zone.dart 1434:47                                          _rootRunUnary
../dart-sdk/lib/async/zone.dart 1335:19                                          runUnary
../dart-sdk/lib/async/future_impl.dart 160:18                                    handleValue
../dart-sdk/lib/async/future_impl.dart 767:44                                    handleValueCallback
../dart-sdk/lib/async/future_impl.dart 796:13                                    _propagateToListeners
../dart-sdk/lib/async/future_impl.dart 602:5                                     [_completeWithValue]
../dart-sdk/lib/async/future_impl.dart 640:7                                     callback
../dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
../dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
../dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15           <fn>

The test description was:
  text selection style 2
════════════════════════════════════════════════════════════════════════════════════════════════════

09:25 +913 ~16 -1: test/material/text_field_test.dart: text selection style 2 [E]                                                                                                                      
  Test failed. See exception logs above.
  The test description was: text selection style 2
```

**Comment by mdebbar** [comment]
It looks like the failing image has been [triaged as negative](https://flutter-gold.skia.org/detail?test=material.text_field_golden.TextSelectionStyle.2&digest=7fbe2f57b4d0038bba905ff114d3e61c) about 17 weeks ago. So that image has been failing in post-submit ever-since, but it wasn't being registered as a failure. If you look at any of the previous successful runs of ` Linux web_tests_6 `, you'll find this in the logs:
```
<<< goldctl imgtest add stderr: Test: material.text_field_golden.TextSelectionStyle.2 FAIL
```

As of yesterday, this [commit](https://github.com/flutter/flutter/commit/ea5201389503be8b5d6b87ccd133502b26924762) started registering these as failures in post-submit. I think the PR is working as intended, it's letting us know that something is going wrong in post-submit.

@Piinks what do you suggest here to unblock the tree?

**Comment by Piinks** [comment]
Looking at the history of the test, there does not appear to have ever been a positive version of this image for web.

This golden file test was introduced in #85640 by @Renzo-Olivares 

These are marked positive for linux/mac/windows:

https://flutter-gold.skia.org/search?corpus=flutter&include_ignored=false&left_filter=name%3Dwidgets.selectable_text_golden.TextSelectionStyle.2&max_rgba=0&min_rgba=0&negative=true&not_at_head=false&positive=true&reference_image_required=false&right_filter=&sort=descending&untriaged=true

And the web version is marked negative:
https://flutter-gold.skia.org/detail?test=material.text_field_golden.TextSelectionStyle.2&digest=7fbe2f57b4d0038bba905ff114d3e61c

I am guessing since all the other platforms were marked positive, but the web one was marked negative, what was actually intended was to skip the web image result. 

@justinmc can you confirm if this sounds right? I am going to prepare a change that skips this test on web so if that's the case we can land it and unblock the tree. :)

**Comment by Piinks** [comment]
[selectable_text_golden.TextSelectionStyle.2](https://flutter-gold.skia.org/search?corpus=flutter&include_ignored=false&left_filter=name%3Dwidgets.selectable_text_golden.TextSelectionStyle.2&max_rgba=0&min_rgba=0&negative=true&not_at_head=false&positive=true&reference_image_required=false&right_filter=&sort=descending&untriaged=true) is the image in question, but I noticed [selectable_text_golden.TextSelectionStyle.1](https://flutter-gold.skia.org/search?corpus=flutter&include_ignored=false&left_filter=name%3Dmaterial.text_field_golden.TextSelectionStyle.1&max_rgba=0&min_rgba=0&negative=true&not_at_head=false&positive=true&reference_image_required=false&right_filter=&sort=descending&untriaged=true) seems to support web. So it is also possible this was just accidentally marked negative and should be positive.

**Comment by Piinks** [comment]
#93722 will skip that test if that is the appropriate fix, I don't know why the image was marked negative. Will need to wait for confirmation on the intention.

**Comment by mdebbar** [comment]
The web doesn't support `selectionHeightStyle` well. That's probably the reason the image was wrong and was marked as negative for the web. I couldn't find an issue for it, so I created [one](https://github.com/flutter/flutter/issues/93723).

This is probably not going to be implemented for a while since we haven't gotten any requests for it from our customers or users.

**Comment by Piinks** [comment]
Ok thanks @mdebbar! It sounds like a test we would typically skip then on web because it's not currently supported. Most of our skipped tests are exactly that. :)

**Comment by Piinks** [comment]
#93722 did not resolve the failure, which is weird because it should have. The failing test was skipped in #93722, but it is still running and failing on the dashboard.

@godofredoc or @CaseyHillers, is it possible the dashboard has the wrong info?

**Comment by Piinks** [comment]
MY bad, it looks like we skipped the wrong test. Opening a new PR now.

**Comment by Piinks** [comment]
(The real) fix in in https://github.com/flutter/flutter/pull/93740 :)

**Comment by CaseyHillers** [comment]
> My bad, it looks like we skipped the wrong test. Opening a new PR now.

The luci scheduler will greedily just test in batches of 3. So it's likely one of the gray rows was the culprit. It doesn't implement any bisection logic.
